gensound/transforms.py

```diff
@@ -187,34 +187,49 @@ def realise(self, audio):
 
 ######### Level/ampltidue Stuff ###################
 
-class Fade(Transform):
-    min_fade = -50
-    
-    def __init__(self, is_in=True, duration=3e3):
-        self.is_in = is_in
+class Fade(Transform):   
+    def __init__(self, duration=None, curve=None, is_in=True, degree=1):
+        assert curve in ("linear", "polynomial")
+        # default setting for backwards compatibility (allowing usage of both Fade() and FadeIn/FadeOut()
+        if curve == None: curve = "linear"
+        if duration == None: duration = 3e3
+
+        self.curve = curve
         self.duration = duration
-    
+        self.is_in = is_in
+        self.degree = degree
+
     def realise(self, audio):
-        amp = DB_to_Linear(np.linspace(Fade.min_fade, 0, self.num_samples(audio.sample_rate)))
-        #amp = np.linspace(0, 1, self.num_samples(audio.sample_rate))
+        #curve type handler
+        if self.curve == "linear":
+            amp = np.linspace(0, 1, self.num_samples(audio.sample_rate))
+        elif self.curve == "polynomial":
+            amp = (np.linspace(0, 1, self.num_samples(audio.sample_rate))) ** self.degree      
+        
+        # fade in/out  handler
+        if self.is_in: audio.audio[:,:len(amp)] *= amp
+        else: audio.audio[:,-len(amp):] *=  amp[::-1]
+
         # perhaps the fade in should be nonlinear
         # TODO subsciprability problem
         
-        if self.is_in:
-            audio.audio[:,:len(amp)] *= amp
-        else:
-            audio.audio[:,-len(amp):] *= amp[::-1]
-        
         # TODO in case of fade out, if amp is shorter or longer than audio,
         # care must be taken when multiplying!
         # TODO TEST!!!!!!!!!!!!!
         # TODO I still hear a bump when the playback starts
 
+class FadeIn(Fade):
+    def __init__(self, duration=None, curve=None, is_in=True, degree=1):
+        super().__init__(duration, curve, is_in,degree)
+
```

Comment:
I think you can now appreciate the advantage of having `FadeIn` and `FadeOut` inherit from `Fade`: this code is shorter than the previous, and easier to maintain.

However, I'm confused as to why we have the `is_in` argument here in the first place?